app/controllers/api/v4/teleconsultation_medical_officers_controller.rb

```diff
@@ -0,0 +1,29 @@
+class Api::V4::TeleconsultationMedicalOfficersController < APIController
+  def sync_to_user
+    medical_officers = current_facility_group.facilities.map { |facility|
+      facility_medical_officers(facility)
+    }
+    render json: to_response(medical_officers)
+  end
+
+  private
+
+  def facility_medical_officers(facility)
+    {id: facility.id,
+     facility_id: facility.id,
+     medical_officers: transform_medical_officers(facility.teleconsultation_medical_officers),
+     created_at: Time.current,
+     updated_at: Time.current,
+     deleted_at: Time.current}
```

Comment:
It seems a bit inconsistent that this endpoint would return a nested array of objects, each containing an array of MOs. Instead, I think we should return a flat array of medical officer objects, each of which contains a `facility_id` in its attributes. What do you think?

Reply:
There's two reasons to make this nested:
- MOs are not implemented as a "real resource", both on the server and the app. This resource represents only a relation between the facility and users. Both a flat resource and a nested resource would need the same effort to serialize/deserialize.
- The payload size becomes much smaller when we nest.

Another additional (later discovered) advantage of this is that we can send all these facility records in one response. The number of flattened records will be much larger and we will have to chunk it, which is slightly tricky. Nesting solves this problem.

Reply:
> MOs are not implemented as a "real resource"

Actually, this may be incorrect. Given that we are sending the name and phone number along, the app _might_ be constructing a MO model internally. There is no need for it to be associating these to `users`. Regardless, nesting still has those advantages.

This data should ideally be in the facility's payload, yes (conditionally added after login). This is a separate endpoint only because it's more ergonomic for the app this way.

Calling it `teleconsultation_facilities` makes sense, although `teleconsultation_medical_officers` appears more illustrative of its purpose (this endpoint is primarily to get MO data on the app). wdyt?

Reply:
Haha yeah tricky. I think I'd slightly lean towards calling it teleconsultation facilities. Reason being: Even though we do primarily want MO info, I'd place a little more weight on ensuring the top-level objects returned by the API match its name. For example, I wouldn't expect a `facilities` endpoint to return facility _groups_, with facility info inside.

Reply:
Could also call it `FacilityMedicalOfficers`

Reply:
Changed it to `/v4/facility_medical_officers/sync` in d7409f6.